core/os/keyboard.h

```diff
@@ -249,7 +249,7 @@ enum class Key {
 
 enum class KeyModifierMask {
 	CODE_MASK = ((1 << 23) - 1), ///< Apply this mask to any keycode to remove modifiers.
-	MODIFIER_MASK = (0x7F << 22), ///< Apply this mask to isolate modifiers.
+	MODIFIER_MASK = (0x7E << 22), ///< Apply this mask to isolate modifiers.
```

Comment:
> Currently the upper 9 bits are reserved for modifiers

Not sure if `KPAD` and `GROUP_SWITCH` should be considered modifiers.

But both seems to be unused:
- `GROUP_SWITCH` flag seems to be completely unused (I have no idea what it was intended for).
- `KPAD` flag is parsed by macOS native menu code, but it must an oversight, since it's never set by any code.

I would probably mask it like this:

```
0-20  Unicode value       |1|
21    Reserved            |1| CODE_MASK
22    Special key flag    |1|
23    Reserved                 |2| Keep reserved in case we want to add some other special flag.
24    CTRL or CMD flag              |3|
25    SHIFT flag                    |3|
26    ALT flag                      |3|
27    META flag                     |3| MODIFIER_MASK
28    CTRL flag                     |3|
29    (KPAD) Reserved               |3|
30    (GROUP_SWITCH) Reserved       |3|
31    Reserved                           |4| Sign bit, keep clean to avoid negative values if it's used with int32_t API.
```

Reply:
Sounds good, so that would boil down to:
```suggestion
	MODIFIER_MASK = (0x7F << 24), ///< Apply this mask to isolate modifiers.
```

Reply:
@Yelloween10 Please see the above suggestion.

Reply:
> @Yelloween10 Please see the above suggestion.

I was just working on it at the moment. I was busy for a few weeks, I have time to do it now.